### DISCUSSION TITLE
Next.JS / Now and static site generator

### DISCUSSION THREAD

**Discussion by fveauvy** [vercel/next.js#11952] [Help]
Hi everyone !

I'm trying to setup a static documentation with authentication.
I use [Mkdocs](mkdocs.org) for building my doc (its a static site generator). The output is a static website containing html, css and js files. Until now I handled authentication with an Express server.

As I'm a big fan of Zeit/Now I would like to use it for deployment. 
I first tried to deploy an Express server alongside my static folder on Now. It worked fine on localhost but I was unable to fix some issues on preview deployment with Now. After reading a lot of discussions on Spectrum, I learnt that Express wasn't a great option if I wanted to deploy serverless because of how lambda function works.

Then I decided to try Next.js because of the API routes and static file serving features. 
Thanks to the awesome "next-connect and Passport" example I could manage authentication. However I'm not sure if what I'm trying to achieve is the proper way or even possible.

 - Is it possible to prevent a user to access public assets served by nextjs (while he's not authenticated) ?
 - Is it possible that Next.js looks for the static HTML file based on request. For example, with the route "/docs/about", it will serve "/docs/about/index.html" ?

I would be grateful for any advice or comment on the matter :)

**Comment by paulogdm** [comment]
Hello @fveauvy ,

Answering your question: It is not possible to prevent access to public files, at least not with ZEIT Now. However, we are planning to release "deployment authentication" very soon, and it is already being tested internally. If you are interested in this feature, reach out to sales@zeit.co so we can understand your use-case better.
